### PR TITLE
Cartesian controller - Fixed a bug

### PR DESCRIPTION
**Pull request by MaChristoph**
Fixed a bug in the cartesian_controller.cpp. 

We used to start the tracking_service of the frame_tracker before the "target_frame" was published for the first time. This doesn't work anymore, so I fixed the serviceCall function. Additionally, I added a readme file for the lwa4p example.launch and another launch file for the cob4-2 simulation.

### CODE REVIEW

File: cob_cartesian_controller/src/cartesian_controller.cpp [hunk 1]
```diff
@@ -177,6 +174,9 @@ bool CartesianController::posePathBroadcaster(const geometry_msgs::PoseArray& ca
 
         tf_broadcaster_.sendTransform(tf::StampedTransform(transform, ros::Time::now(), cartesian_path.header.frame_id, target_frame_));
 
+        // Activate Tracking
+        startTracking(); // Somehow the behavior changed.. we used to call this one before the first pose was published but this doesn't work anymore.
+
```
Comment (benmaidel) [inline cob_cartesian_controller/src/cartesian_controller.cpp:179]: with this you are calling the start_tracking service for every pose on the path. Is this really intended? You probably just have to take care that the transform specified in the path is available before calling tracking service.

File: cob_obstacle_distance/src/distance_manager.cpp [hunk 2]
```diff
@@ -90,7 +90,7 @@ int DistanceManager::init()
         return -1;
     }
 
-    if(!nh_.getParam("joint_names", this->joints_))
+    if(!nh_.getParam("/arm_left/joint_names", this->joints_))
```
Comment (benmaidel) [inline cob_obstacle_distance/src/distance_manager.cpp:93]: do not use absolute namespaces! With this the distance_manager is just usable with the left arm...

File: cob_twist_controller/include/cob_twist_controller/constraints/constraint_ca_impl.h [hunk 3]
```diff
@@ -389,22 +393,26 @@ void CollisionAvoidance<T_PARAMS, PRIO>::calcPredictionValue()
 
     if (params.frame_names.end() != str_it)
     {
+        unsigned int dof;
         if (this->constraint_params_.current_distances_.size() > 0)
         {
             uint32_t frame_number = (str_it - params.frame_names.begin()) + 1;  // segment nr not index represents frame number
             KDL::FrameVel frame_vel;
-
+            ROS_WARN_STREAM("frame_number" << frame_number);
```
Comment (benmaidel) [inline cob_twist_controller/include/cob_twist_controller/constraints/constraint_ca_impl.h:401]: I suppose you used this warning for debugging purposes. Please remove such debugging outputs before PR

File: cob_twist_controller/src/cob_twist_controller.cpp [hunk 4]
```diff
@@ -142,7 +142,8 @@ bool CobTwistController::initialize()
     {
         twist_controller_params_.frame_names.push_back(chain_.getSegment(i).getName());
     }
-    register_link_client_ = nh_.serviceClient<cob_srvs::SetString>("obstacle_distance/registerLinkOfInterest");
+
+    register_link_client_ = nh_.serviceClient<cob_srvs::SetString>("/register_links");
```
Comment (benmaidel) [inline cob_twist_controller/src/cob_twist_controller.cpp:146]: Do not use absolute namespaces here. You can remap the namespaces inside the launch file if you need to.

File: cob_twist_controller/src/cob_twist_controller.cpp [hunk 5]
```diff
@@ -164,7 +165,7 @@ bool CobTwistController::initialize()
     ros::Duration(1.0).sleep();
 
     /// initialize ROS interfaces
-    obstacle_distance_sub_ = nh_.subscribe("obstacle_distance", 1, &CallbackDataMediator::distancesToObstaclesCallback, &callback_data_mediator_);
+    obstacle_distance_sub_ = nh_.subscribe("/obstacle_distances", 1, &CallbackDataMediator::distancesToObstaclesCallback, &callback_data_mediator_);
```
Comment (benmaidel) [inline cob_twist_controller/src/cob_twist_controller.cpp:168]: do not use absolute namespaces! With this the distance_manager is just usable with the left arm...

File: cob_twist_controller/src/cob_twist_controller.cpp [hunk 6]
```diff
@@ -309,7 +310,7 @@ void CobTwistController::checkSolverAndConstraints(cob_twist_controller::TwistCo
     {
         if (!register_link_client_.exists())
         {
-            ROS_ERROR("ServiceServer 'obstacle_distance/registerLinkOfInterest' does not exist. CA not possible");
+            ROS_ERROR("ServiceServer '/register_links' does not exist. CA not possible");
```
Comment (benmaidel) [inline cob_twist_controller/src/cob_twist_controller.cpp:313]: do not use absolute namespaces! With this the distance_manager is just usable with the left arm...

File: cob_cartesian_controller/src/cartesian_controller.cpp [hunk 7]
```diff
@@ -355,3 +355,14 @@ void CartesianController::actionAbort(const bool success, const std::string& mes
     as_->setAborted(action_result_, action_result_.message);
     stopTracking();
 }
+
+bool CartesianController::initTrackingFrame()
```
Comment (benmaidel) [inline cob_cartesian_controller/src/cartesian_controller.cpp:359]: you need to declare the member fkt inside the header

**Comment by MaChristoph** [comment]
@ipa-bnm 
Oh there went something wrong.. I only wanted to pull request the changes in my cob_cartesian_controller. The other stuff is "work in progress"

**Comment by mgruhler** [comment]
new PR will come soon....

**Pull request closed without merging**